src/wp-includes/rest-api/endpoints/class-wp-rest-block-navigation-areas-controller.php

```diff
@@ -0,0 +1,285 @@
+	 */
+	public function get_item( $request ) {
+		$name = $request['area'];
+		$area = $this->get_navigation_area_object( $name );
+		$data = $this->prepare_item_for_response( $area, $request );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Updates a specific navigation area.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function update_item( $request ) {
+		$name = $request['area'];
+
+		$mapping          = get_option( 'fse_navigation_areas', array() );
+		$mapping[ $name ] = $request['navigation'];
+		update_option( 'fse_navigation_areas', $mapping );
+
+		$area = $this->get_navigation_area_object( $name );
+		$data = $this->prepare_item_for_response( $area, $request );
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Converts navigation area name to a convenient object that this endpoint can reason about.
+	 *
+	 * @param string $name Navigation area name.
+	 * @return stdClass An object representation of the navigation area.
+	 */
+	private function get_navigation_area_object( $name ) {
+		$available_areas   = get_navigation_areas();
+		$mapping           = get_option( 'fse_navigation_areas', array() );
+		$area              = new stdClass();
+		$area->name        = $name;
+		$area->navigation  = ! empty( $mapping[ $name ] ) ? $mapping[ $name ] : null;
+		$area->description = $available_areas[ $name ];
+		return $area;
+	}
+
+	/**
+	 * Prepares a navigation area object for serialization.
+	 *
+	 * @param stdClass        $area Post status data.
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_REST_Response Post status data.
+	 */
+	public function prepare_item_for_response( $area, $request ) {
+		$areas      = get_navigation_areas();
+		$navigation = ( isset( $areas[ $area->name ] ) ) ? $area->navigation : 0;
+
+		$fields = $this->get_fields_for_response( $request );
+		$data   = array();
+
+		if ( rest_is_field_included( 'name', $fields ) ) {
+			$data['name'] = $area->name;
+		}
+
+		if ( rest_is_field_included( 'description', $fields ) ) {
+			$data['description'] = $area->description;
+		}
+
+		if ( rest_is_field_included( 'navigation', $fields ) ) {
+			$data['navigation'] = (int) $navigation;
+		}
+
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		$response = rest_ensure_response( $data );
+
+		/**
+		 * Filters a navigation area returned from the REST API.
+		 *
+		 * Allows modification of the navigation area data right before it is
+		 * returned.
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param object $area The original status object.
+		 * @param WP_REST_Request $request Request used to generate the response.
```

Comment:
These need to be aligned, https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#1-functions-class-methods